### ISSUE TITLE
Document SCALE Optimizations for Integer Types and Vectors Thereof

### ISSUE THREAD

**Issue by danforbes** [substrate-developer-hub/knowledgebase#43]
Document the concepts mentioned by @thiolliere in [comments on #38](https://github.com/substrate-developer-hub/knowledgebase/pull/38#discussion_r411317372).

**Comment by gui1117** [comment]
Indeed maybe mentionning those optimization for the rust implementation (and so the one used in FRAME) of parity-scale-codec is interesting.
Also I guess we should document them in the code as well.

**Comment by bkchr** [comment]
Okay, we speak here about the internal optimizations of the implementation? Why should they be documented here? 

This is nothing a user should care about, this is just an optimization of the implementation.

**Comment by danforbes** [comment]
I tend to agree with @bkchr. I think it is very important to be careful that we don't document too much because it will just confuse our users. Internal optimizations definitely fall into the realm of "too much" for me.
